core/modules/file.py

```diff
@@ -239,6 +239,35 @@ class File(Tree):
 
 	blobCacheTime = 60 * 60 * 24  # Requests to file/download will be served with cache-control: public, max-age=blobCacheTime if set
 
+	def write(self, filename: str, content: Any, mimetype: str = "text/plain"):
+		"""
+		Write a file from any buffer into the file module.
+
+		:param filename: Filename to be written.
+		:param content:  The file content to be written, as bytes-like object.
+		:param mimetype: The file's mimetype.
+
+		:return: Returns the key of the file object written. This can be associated e.g. with a fileBone.
+		"""
+		dl_key = utils.generateRandomString()
+
+		blob = bucket.blob("%s/source/%s" % (dl_key, filename))
+		blob.upload_from_file(BytesIO(content), content_type=mimetype)
+
+		skel = self.addSkel("leaf")
+		skel["name"] = filename
+		skel["size"] = blob.size
+		skel["mimetype"] = mimetype
+		skel["dlkey"] = dl_key
+		skel["parentdir"] = None
+		skel["parententry"] = None
+		skel["pending"] = False
+		skel["weak"] = True
+		skel["width"] = 0
+		skel["height"] = 0
```

Comment:
This is wrong. If you set this to `0` it means "it's has a value" (which is 0), but this isn't determined correctly according to the content here. 
Height and width are currently only used for images in the core. But you can implement this for different file-types as well. For instance the height of a text-file could be the amount of lines and the width the amount of columns, for a pdf you can use the dimension of the page(s). Therefor this attributes should be flexible and always contain a reliable int (or nothing).

My suggestion: Extend the signature with kwargs like `height: Optional[int] = None, width: Optional[int] = None`. Then you don't have a value by default (= `None`) and you can pass a correct value with the `File.write()` call.


```suggestion
		skel["width"] = width   # from the signature 
		skel["height"] = height   # from the signature
```